### PR TITLE
Display error message when build script is not generated

### PR DESCRIPTION
**Pull request by BanzaiMan**
When users' `.travis.yml` has problems with the data structure (rather than invalid YAML), `travis-build` just generally throws an exception, and we display a generic error message:

    An error occurred while generating the build script.

(e.g., https://travis-ci.org/gligoran/cordova-set-version/jobs/223634347#L1)

This is not very helpful. Here, we `rescue` the Exception, and give some advice if we can.

### CODE REVIEW

**Comment by BanzaiMan** [comment]
Now has backtrace in a fold. https://staging.travis-ci.org/BanzaiMan/travis_staging_test/builds/574118#L11

File: lib/travis/build/script.rb [hunk 1]
```diff
@@ -240,6 +246,37 @@ def debug_enabled?
...
+          if exception.is_a? Travis::Build::CompilationError
+            msg = [
+              exception.message
+            ]
+            doc_path = exception.doc_path
+          else
+            msg = [
+              "Unfortunately, we do not know much about this error.",
+              "",
+              "#{exception.class}: #{exception.message}"
```
Comment (meatballhat) [inline lib/travis/build/script.rb]: Is there any risk of secret leakage in `exception.message`?

**Comment by BanzaiMan** [comment]
Examples:

* [a generic error](https://staging.travis-ci.org/BanzaiMan/travis_staging_test/builds/577974)
* [`addons.apt` definition error](https://staging.travis-ci.org/BanzaiMan/travis_staging_test/builds/577709)
* [`deploy` config error](https://staging.travis-ci.org/BanzaiMan/travis_staging_test/builds/574118)

File: lib/travis/build/script.rb (hunk 1 above)
Comment (BanzaiMan) [inline lib/travis/build/script.rb]: Here, `exception` is a builtin Ruby exception. I do not believe it will contain user input here.

Comment (BanzaiMan) [inline lib/travis/build/script.rb]: Well, that's not necessarily true. In one example, it actually comes from RubyGems' argument error.

Without the error message, the argument class is useless. :-(

**Comment by BanzaiMan** [comment]
We will display the Sentry error ID instead. https://staging.travis-ci.org/BanzaiMan/travis_staging_test/builds/578842#L11

**Pull request merged**